src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -283,6 +283,25 @@ protected BoundExpression LowerTest(BoundDagTest test)
                         throw ExceptionUtilities.UnexpectedValue(test);
                 }
             }
+
+            private BoundExpression MakeNullCheck(SyntaxNode syntax, BoundExpression rewrittenExpr, BinaryOperatorKind operatorKind)
+            {
+                if (rewrittenExpr.Type.IsPointerOrFunctionPointer())
+                {
+                    TypeSymbol objectType = _factory.Compilation.GetSpecialType(SpecialType.System_Object);
```

Comment:
>GetSpecialType [](start = 65, length = 14)

We shouldn't be requesting helpers, types or methods, from compilation without checking success of the result. And likely should recover in case of failures. I believe ```_factory``` itself has helper methods to lookup helpers that report appropriate diagnostics and probably throw to avoid need for any special recovery in case of a failure.